### PR TITLE
Fix 1.9 syntax error

### PR DESCRIPTION
**Pull request by BanzaiMan**
## Description
0.1x is supposed to be 1.9-compat, so change this method signature to reflect it.

## Todos
List any remaining work that needs to be done, i.e:
- [ ] Tests
- [ ] Documentation

## Additional Notes
https://travis-ci.com/travis-ci/travis.rb/jobs/263075645#L271

### CODE REVIEW

**Comment by BanzaiMan** [comment]
Not sure how to fix the CI.

**Comment by olleolleolle** [comment]
@BanzaiMan GitHub actions doesn't support 2.3 any longer. On master, that element of the ci matrix was dropped, to go green.

Such a PR is needed in this branch, too.

**Comment by BanzaiMan** [comment]
Not sure why GitHub thinks 2.3 is still required and expected. ¯\_😕_/¯

File: lib/faraday/error.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def initialize(exc = 'timeout', response = nil)
 
   # Raised by Faraday::Response::RaiseError in case of a nil status in response.
   class NilStatusError < ServerError
-    def initialize(_exc, response: nil)
+    def initialize(_exc, response = nil)
```
Comment (iMacTia) [inline lib/faraday/error.rb]: This is NOT the same thing as `def initialize(_exc, response: nil)`

**Comment by iMacTia** [comment]
Well, according to the 0.1x README we do 😃! But no tests are executed against 1.9 anymore, so hard to catch this sort of things 🤔

File: lib/faraday/error.rb (hunk 1 above)
Comment (BanzaiMan) [reply lib/faraday/error.rb]: oh, that's true. However, I will point out that this is the same call signature as that of `TimeoutError` on line 80. If this is not the right way, then `TimeoutError` should also be revisited.

**Comment by BanzaiMan** [comment]
https://github.com/lostisland/faraday/pull/1086/files#diff-1b35e95f85a819bcd3c16d920ecdcffeR112-R113 has `%i`, which I believe is not 1.9-compatible.

File: lib/faraday/error.rb (hunk 1 above)
Comment (mrexox) [reply lib/faraday/error.rb]: Can it be rewritten to accept a hash parameter? Like that:
```ruby
def initialize(_exc, params)
  response = params[:response]
```

**Comment by iMacTia** [comment]
Unfortunately I was unable to get 1.9 working again on our GitHub Actions CI, so it's hard to say if we can or can't merge this PR.
Moreover we've consistently merged incompatible changes with 2.0, possibly more than the ones already highlighted in this PR.

@BanzaiMan are you still using Ruby 1.9 together with the latest Faraday version?
@olleolleolle @technoweenie latest releases (0.17.x, possibly sooner) have already broken the compatibility with 1.9, should we come back on track or just drop the official support?

**Comment by BanzaiMan** [comment]
@iMacTia  > are you still using Ruby 1.9 together with the latest Faraday version?
Yes. For many reasons, we need Faraday to provide a 1.9-compatible library in some form. I understand that 1.9 is certifiably EOL, but if it is no longer supported, the incompatible versions should be yanked so that we have some reasonable version that supports 1.9 can be found.

Looks to me that #1094 supersedes this, so I'll close this now.

**Pull request closed without merging**